pkg/controllermanager/controller/event/event.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package event
+
+import (
+	"context"
+	"github.com/gardener/gardener/pkg/client/kubernetes/clientmap"
+	"github.com/gardener/gardener/pkg/client/kubernetes/clientmap/keys"
+	"github.com/gardener/gardener/pkg/controllermanager"
+	"github.com/gardener/gardener/pkg/controllermanager/apis/config"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	"github.com/gardener/gardener/pkg/logger"
+	"github.com/prometheus/client_golang/prometheus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sync"
+	"time"
+)
+
+// Controller controls Events.
+type Controller struct {
+	clientMap              clientmap.ClientMap
+	cfg                    *config.ControllerManagerConfiguration
+	eventQueue             workqueue.RateLimitingInterface
+	numberOfRunningWorkers int
+	workerCh               chan int
+}
+
+// NewController instantiates a new event controller
+func NewController(clientMap clientmap.ClientMap, cfg *config.ControllerManagerConfiguration) *Controller {
```

Comment:
I know, the other controllers also do it this way. But WDYT about passing only the required `EventControllerConfiguration`? (only a suggestion, no strong opinion)
```suggestion
func NewController(clientMap clientmap.ClientMap, cfg *config.EventControllerConfiguration) *Controller {
```

Reply:
Great Idea 🤩